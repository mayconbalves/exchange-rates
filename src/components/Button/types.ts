export type ButtonProps = {
  disabled?: boolean
  onClick: () => void
  title: string
}
