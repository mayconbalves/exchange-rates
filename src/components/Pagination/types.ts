export type PaginationProps = {
  currentPage: number
  itemsPerPage: number
  totalItems: number
  onPageChange: (page: number) => void
}
