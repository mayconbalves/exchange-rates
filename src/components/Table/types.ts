export type TableDate = {
  rate: string
  value: number
  date: string
  timestamp: number
}

export type SortConfig = {
  key: keyof TableDate | null
  direction: 'ascending' | 'descending'
}

export type TableProps = {
  rates: TableDate[]
  date: string
}
