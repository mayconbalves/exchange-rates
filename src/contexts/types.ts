export type ToastContextProps = {
  showToast: (message: string, type: 'success' | 'error') => void
}
