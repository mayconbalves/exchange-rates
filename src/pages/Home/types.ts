export type Registrations = {
  id: string
  admissionDate: string
  email: string
  employeeName: string
  status: 'REPROVED' | 'APPROVED' | 'REVIEW'
  cpf: string
}
