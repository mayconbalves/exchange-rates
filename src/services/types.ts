export type RateItem = {
  date: string
  rate: string
  value: number
  timestamp: number
}

export type ApiResponse = {
  success: boolean
  timestamp: number
  base: string
  date: string
  rates: Record<string, number>
}
