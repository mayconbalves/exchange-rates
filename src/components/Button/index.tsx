import styles from './styles.module.css'
import { ButtonProps } from './types'

const Button = ({ disabled, onClick, title }: ButtonProps) => {
  return (
    <button className={styles.button} onClick={onClick} disabled={disabled}>
      {title}
    </button>
  )
}

export default Button
