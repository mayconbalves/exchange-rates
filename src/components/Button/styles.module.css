.button {
  background-color: #ffd862;
  border: 1px solid #000;
  border-radius: 4px;
  cursor: pointer;
  font-size: medium;
  height: 40px;
  margin: 0 10px;
  padding: 5px 10px;
  width: 100px;
}

.button:disabled {
  border: none;
  background-color: #ccc;
  cursor: not-allowed;
}

.button:hover:enabled {
  background-color: #fee28e;
  box-shadow: 1px 1px 1px rgba(0, 0, 0, 0.8);
}
