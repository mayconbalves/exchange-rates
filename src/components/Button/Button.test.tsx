import { fireEvent, render, screen } from '@testing-library/react'
import Button from '.'
import { ButtonProps } from './types'

describe('Button Component', () => {
  const defaultProps: ButtonProps = {
    title: 'Clique Aqui',
    onClick: jest.fn(),
    disabled: false
  }

  it('Should be render button title', () => {
    render(<Button {...defaultProps} />)
    expect(screen.getByText('Clique Aqui')).toBeInTheDocument()
  })

  it('Should be call onClick event', () => {
    render(<Button {...defaultProps} />)
    const button = screen.getByRole('button')
    fireEvent.click(button)
    expect(defaultProps.onClick).toHaveBeenCalledTimes(1)
  })

  it('Should be render disabled button', () => {
    render(<Button {...defaultProps} disabled={true} />)
    const button = screen.getByRole('button')
    expect(button).toBeDisabled()
  })

  it('Should be add class', () => {
    render(<Button {...defaultProps} />)
    const button = screen.getByRole('button')
    expect(button).toHaveClass('button')
  })

  it('Should be create snapshot', () => {
    const { container } = render(<Button {...defaultProps} />)

    expect(container).toMatchSnapshot()
  })
})
