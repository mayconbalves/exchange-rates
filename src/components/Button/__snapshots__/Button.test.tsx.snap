// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component Should be create snapshot 1`] = `
<div>
  <button
    class="button"
  >
    Clique Aqui
  </button>
</div>
`;
